### PR TITLE
fix(EVS): EVS resource support job status refresh

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:

EVS resource support job status refresh

- This PR has been tested on Huawei Cloud.
- This PR has been tested on G42 Cloud.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/evs' TESTARGS='-run TestAccEvsVolume_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run TestAccEvsVolume_basic -timeout 360m -parallel 4
=== RUN   TestAccEvsVolume_basic
=== PAUSE TestAccEvsVolume_basic
=== CONT  TestAccEvsVolume_basic
--- PASS: TestAccEvsVolume_basic (452.90s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       452.960s
```

```
make testacc TEST='./huaweicloud/services/acceptance/evs' TESTARGS='-run TestAccEvsVolume_withEpsId'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run TestAccEvsVolume_withEpsId -timeout 360m -parallel 4 
=== RUN   TestAccEvsVolume_withEpsId 
=== PAUSE TestAccEvsVolume_withEpsId 
=== CONT  TestAccEvsVolume_withEpsId 
--- PASS: TestAccEvsVolume_withEpsId (47.67s) 
PASS 
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       47.723s
```

```
make testacc TEST='./huaweicloud/services/acceptance/evs' TESTARGS='-run TestAccEvsVolume_prePaid'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run TestAccEvsVolume_prePaid -timeout 360m -parallel 4 
=== RUN   TestAccEvsVolume_prePaid 
=== PAUSE TestAccEvsVolume_prePaid
=== CONT  TestAccEvsVolume_prePaid
--- PASS: TestAccEvsVolume_prePaid (155.53s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       155.604s
```

- G42 unit test
```
make testacc TEST='./g42cloud' TESTARGS='-run TestAccEvsStorageV3Volume_'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./g42cloud -v -run TestAccEvsStorageV3Volume_ -timeout 360m -parallel=4 
=== RUN   TestAccEvsStorageV3Volume_basic 
=== PAUSE TestAccEvsStorageV3Volume_basic 
=== RUN   TestAccEvsStorageV3Volume_image
=== PAUSE TestAccEvsStorageV3Volume_image
=== CONT  TestAccEvsStorageV3Volume_basic
=== CONT  TestAccEvsStorageV3Volume_image
--- PASS: TestAccEvsStorageV3Volume_image (55.54s) 
--- PASS: TestAccEvsStorageV3Volume_basic (60.52s) 
PASS
ok      github.com/g42cloud-terraform/terraform-provider-g42cloud/g42cloud      60.569s
```

### CODE REVIEW

File: huaweicloud/services/evs/resource_huaweicloud_evs_volume.go [hunk 1]
```diff
@@ -289,6 +291,19 @@ func resourceEvsVolumeCreate(ctx context.Context, d *schema.ResourceData, meta i
 		}
 	}
 
+	// If charging mode is postPaid, wait for the job to be SUCCESS
```
Comment (Lance52259) [inline huaweicloud/services/evs/resource_huaweicloud_evs_volume.go]:
```suggestion
	// If charging mode is postPaid, wait for the job status to SUCCESS
```

File: huaweicloud/services/evs/resource_huaweicloud_evs_volume.go [hunk 2]
```diff
@@ -307,6 +322,36 @@ func resourceEvsVolumeCreate(ctx context.Context, d *schema.ResourceData, meta i
...
+	_, err := stateConf.WaitForStateContext(ctx)
+	return err
+}
+
+func evsJobRefreshFunc(c *golangsdk.ServiceClient, jobId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := evsJobs.GetJobDetails(c, jobId).ExtractJob()
+		if err != nil {
+			// there has no specify code here
+			return resp, "ERROR", err
```
Comment (houpeng80) [inline huaweicloud/services/evs/resource_huaweicloud_evs_volume.go:344]: it is no need to set state if err is not nil, it is same when the value of state is ERROR or empty

File: huaweicloud/services/evs/resource_huaweicloud_evs_volume.go (hunk 1 above)
Comment (deer-hang) [reply huaweicloud/services/evs/resource_huaweicloud_evs_volume.go]: OK

File: huaweicloud/services/evs/resource_huaweicloud_evs_volume.go (hunk 2 above)
Comment (deer-hang) [reply huaweicloud/services/evs/resource_huaweicloud_evs_volume.go:344]: OK, I have remove it.

File: huaweicloud/services/evs/resource_huaweicloud_evs_volume.go [hunk 3]
```diff
@@ -307,6 +322,36 @@ func resourceEvsVolumeCreate(ctx context.Context, d *schema.ResourceData, meta i
...
+	_, err := stateConf.WaitForStateContext(ctx)
+	return err
+}
+
+func evsJobRefreshFunc(c *golangsdk.ServiceClient, jobId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := evsjobs.GetJobDetails(c, jobId).ExtractJob()
+		if err != nil {
+			// there has no special code here
+			return resp, "", err
```
Comment (Lance52259) [inline huaweicloud/services/evs/resource_huaweicloud_evs_volume.go]:
```suggestion
			return resp, "ERROR", err
```
the empty string will make the customer confuse,: `expect 'SUCCESS', but ''`

Comment (deer-hang) [reply huaweicloud/services/evs/resource_huaweicloud_evs_volume.go]: Done

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/evs' TESTARGS='-run TestAccEvsVolume_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run TestAccEvsVolume_basic -timeout 360m -parallel 4 
=== RUN   TestAccEvsVolume_basic 
=== PAUSE TestAccEvsVolume_basic
=== CONT  TestAccEvsVolume_basic
--- PASS: TestAccEvsVolume_basic (175.42s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       175.532s
```

File: huaweicloud/services/evs/resource_huaweicloud_evs_volume.go [hunk 4]
```diff
@@ -16,7 +16,9 @@ import (
 	"github.com/chnsz/golangsdk"
 	"github.com/chnsz/golangsdk/openstack/ecs/v1/block_devices"
 	"github.com/chnsz/golangsdk/openstack/ecs/v1/jobs"
+	evsjobs "github.com/chnsz/golangsdk/openstack/evs/v1/jobs"
```
Comment (ShiChangkuo) [inline huaweicloud/services/evs/resource_huaweicloud_evs_volume.go]: it's better to add the alias for "ecs/v1/jobs" package

```suggestion
	ecsjobs "github.com/chnsz/golangsdk/openstack/ecs/v1/jobs"
	"github.com/chnsz/golangsdk/openstack/evs/v1/jobs"
```

Comment (deer-hang) [reply huaweicloud/services/evs/resource_huaweicloud_evs_volume.go]: Done

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/evs' TESTARGS='-run TestAccEvsVolume_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run TestAccEvsVolume_basic -timeout 360m -parallel 4 
=== RUN   TestAccEvsVolume_basic 
=== PAUSE TestAccEvsVolume_basic
=== CONT  TestAccEvsVolume_basic
--- PASS: TestAccEvsVolume_basic (123.00s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       123.048s
```

**Pull request merged**